src/lib/util/nbo.h

```diff
@@ -145,7 +146,8 @@ static inline uint16_t fr_nbo_to_uint16(uint8_t const data[static sizeof(uint16_
  */
 static inline uint32_t fr_nbo_to_uint24(uint8_t const data[static 3])
 {
-	return (((uint32_t)data[0]) << 16) | (((uint32_t)data[1]) << 8) | data[2];
+	uint8_t const *d = data;
```

Comment:
Can you add a comment saying why this is done.

Reply:
Will do.